transcom/mymove
# B 20441 int

## [Agility ticket](tbd)

## Summary

Removing and replacing a bad migration file for updating unpriced MS and CS service items. 

[this article](tbd) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

I set all of the locked_price_cents and pricing_estimate  with MS and CS codes to null in my local with:
"UPDATE mto_service_items AS ms
SET locked_price_cents = null,
    pricing_estimate = null
FROM re_task_order_fees AS tf
JOIN re_services AS s
ON tf.service_id = s.id
JOIN re_contract_years AS cy
ON tf.contract_year_id = cy.id
JOIN re_contracts AS c
ON cy.contract_id = c.id
WHERE (s.code = 'MS' OR s.code = 'CS') AND ms.re_service_id = s.id;"

Then I manually ran the migration file in pgAdmin to ensure it ran properly and updated only MS and CS service items that were nulled. I then ran it again to ensure that it updated 0 on concurrent runs.

The following query can be used to pull service items in question:
"SELECT * FROM public.mto_service_items ms
JOIN re_services r ON ms.re_service_id = r.id
WHERE r.code = 'MS' OR r.code = 'CS'"

Be sure to check service items that do not have the MS or CS code.


### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
